react/src/views/CreateBox/CreateBox.test.js

```diff
@@ -0,0 +1,270 @@
+import { Router } from "react-router-dom";
+import { createMemoryHistory } from "history";
+import "mutationobserver-shim";
+import { GraphQLError } from "graphql";
+
+const mocks = [
+  {
+    request: {
+      query: CREATE_BOX,
+      variables: {
+        productId: 2,
+        items: 2,
+        locationId: 2,
+        comments: "",
+        sizeId: 2,
+        qrBarcode: "387b0f0f5e62cebcafd48383035a92a",
+      },
+    },
+    result: {
+      data: {
+        createBox: {
+          id: 555,
+          box_id: 456,
+          product_id: 123,
+          items: 50,
+        },
+      },
+    },
+  },
+];
+
+describe("Renders components", () => {
+  let component = null;
+  beforeEach(() => {
+    const history = createMemoryHistory();
+    const state = { qr: "barcode=387b0f0f5e62cebcafd48383035a92a" };
+    history.push("/create-box", state);
+
+    component = render(
+      <MockedProvider mocks={mocks} addTypename={false}>
+        <Router history={history}>
+          <CreateBox />
+        </Router>
+      </MockedProvider>,
+    );
+  });
+
+  afterEach(cleanup);
+
+  it("renders a header titled, `Create a Box`", () => {
+    expect(component.getByRole("heading", { name: /create a box/i })).toHaveTextContent(
+      "Create a Box",
+    );
+  });
+
+  it("renders a form", () => {
+    expect(component.getByTestId("createBoxForm")).toBeTruthy();
+  });
+
+  it("renders the correct 5 fields within the form", () => {
+    expect(component.getByText(/locationid*/i)).toBeTruthy();
+    expect(component.getByText(/productid*/i)).toBeTruthy();
+    expect(component.getByText(/items*/i)).toBeTruthy();
+    expect(component.getByText(/sizeid*/i)).toBeTruthy();
+    expect(component.getByText(/comments*/i)).toBeTruthy();
+  });
+
+  it("renders a submit button titled, `do the mutation`", () => {
+    expect(component.getByRole("button", { name: /do the mutation/i })).toHaveTextContent(
+      "do the mutation",
+    );
+  });
+});
+
+describe("Submission result screen is shown the right way", () => {
+  let component = null;
+  beforeEach(() => {
+    const history = createMemoryHistory();
+    const state = { qr: "barcode=387b0f0f5e62cebcafd48383035a92a" };
+    history.push("/create-box", state);
+
+    component = render(
+      <MockedProvider mocks={mocks} addTypename={false}>
+        <Router history={history}>
+          <CreateBox />
+        </Router>
+      </MockedProvider>,
+    );
+  });
+
+  afterEach(cleanup);
+
+  it("After a successful submission, the form disappears and a screen with `You created a new box` appears with a box-id", async () => {
+    const submitBtn = component.getByRole("button", { name: /do the mutation/i });
+
+    fireEvent.click(submitBtn);
+
+    await waitFor(() => {
+      expect(component.queryByTestId("createBoxForm")).toBeNull();
+      expect(component.getByTestId("createdBox")).toBeTruthy();
+      expect(component.getByRole("heading", { name: /the box id is: 456/i })).toHaveTextContent(
```

Comment:
this part feels weird and looks like duplication
```
{ name: /the box id is: 456/i }
```

Reply:
Would it be more succinct to do this?
```
expect(component.findByRole("heading", { name: /the box id is: 456/i }));
```
This version would omit `.toHaveTextContent()`.
The findBy query returns a resolved Promise when an element is found: https://testing-library.com/docs/queries/about/#types-of-queries

Reply:
yes why not. not a biggy though